src/pages/Dashboard/Markets/BorrowMarket/index.tsx

```diff
@@ -74,19 +76,33 @@ export const BorrowMarketUi: React.FC<IBorrowMarketUiProps> = ({
     ];
   });
 
+  const rowOnClick = (_e: React.MouseEvent<HTMLElement>, row: ITableProps['data'][number]) => {
+    const asset = borrowAssets.find((value: Asset) => value.id === row[0].value);
+    if (asset) {
+      setSelectedAsset(asset);
+    }
+  };
+
   return (
-    <div className={className} css={styles.tableContainer}>
-      <Table
-        title={t('markets.borrowMarketTableTitle')}
-        columns={columns}
-        data={rows}
-        initialOrder={{
-          orderBy: 'apy',
-          orderDirection: 'asc',
-        }}
-        rowKeyIndex={0}
-      />
-    </div>
+    <>
+      <div className={className} css={styles.tableContainer}>
+        <Table
+          title={t('markets.borrowMarketTableTitle')}
+          columns={columns}
+          data={rows}
+          initialOrder={{
+            orderBy: 'apy',
+            orderDirection: 'asc',
+          }}
+          rowOnClick={rowOnClick}
+          rowKeyIndex={0}
+        />
+      </div>
+
+      {selectedAsset && (
```

Comment:
In PR 372 I put each table as its own component and laid them out with modals in the index file.